test/common/upstream/logical_dns_cluster_test.cc

```diff
@@ -115,7 +116,7 @@ TEST_P(LogicalDnsParamTest, ImmediateResolve) {
   )EOF";
 
   EXPECT_CALL(initialized_, ready());
-  EXPECT_CALL(dns_resolver_, resolve("foo.bar.com", std::get<1>(GetParam()), _))
+  EXPECT_CALL(*dns_resolver_, resolve("foo.bar.com", std::get<1>(GetParam()), _))
```

Comment:
I was hoping it would be easier than going all the way down to the `ares_channel`, but I think you are right (the state isn't cached anywhere else). You can add a `getResolverForTest` etc. kinds of methods.